docs/pydoc/loaders.py

```diff
@@ -0,0 +1,63 @@
+import copy
+import typing as t
+
+import docspec
+from pydoc_markdown.interfaces import Context
+from pydoc_markdown.contrib.loaders.python import PythonLoader
+
+
+class CustomPythonLoader(PythonLoader):
+    def load(self) -> t.Iterable[docspec.Module]:
+        """
+        Load the modules, but include inherited methods in the classes.
+        """
+        # Load all haystack modules
+        temp_loader = PythonLoader(search_path=["../../../haystack"])
+        temp_loader.init(Context(directory="."))
+        all_modules = list(temp_loader.load())
+
+        # Collect all classes
+        classes = {}
+        for module in all_modules:
+            for member in module.members:
+                if isinstance(member, docspec.Class):
+                    classes[member.name] = member
+
+        # Load the modules specified in the search path
+        modules = super().load()
+
+        # Add inherited methods to the classes
+        modules = self.include_inherited_methods(modules, classes)
+
+        return modules
+
+    def include_inherited_methods(
+        self, modules: t.Iterable[docspec.Module], classes: t.Dict[str, docspec.Class]
+    ) -> t.Iterable[docspec.Module]:
+        """
+        Recursively include inherited methods from the base classes.
+        """
+        modules = list(modules)
```

Comment:
`modules` is already an Iterable.
Is it necessary to cast it into a List?

Reply:
Yes, this is needed because `modules` is a generator, otherwise `modules` would be empty after the for-loop over it.